libs/iavl/fast_node_cache.go

```diff
@@ -58,37 +58,22 @@ func (fnc *FastNodeCache) uncache(key []byte) {
 // reached the cache size limit.
 func (fnc *FastNodeCache) cache(node *FastNode) {
 	fnc.cacheMutex.Lock()
-	elem := fnc.cacheQueue.PushBack(node)
-	fnc.items[string(node.key)] = elem
 
-	for fnc.cacheQueue.Len() > GetFastNodeCacheSize() {
-		oldest := fnc.cacheQueue.Front()
-		key := fnc.cacheQueue.Remove(oldest).(*FastNode).key
-		delete(fnc.items, amino.BytesToStr(key))
-	}
-	fnc.cacheMutex.Unlock()
-}
-
-func (fnc *FastNodeCache) cacheWithKey(key string, node *FastNode) {
-	fnc.cacheMutex.Lock()
-	elem := fnc.cacheQueue.PushBack(node)
-	fnc.items[key] = elem
+	if elem, ok := fnc.items[string(node.key)]; ok {
+		fnc.cacheQueue.MoveToBack(elem)
```

Comment:
maybe the struct of NodeCache have this problem, its may cause a misunderstanding ？ I think  its good for  both of them coincide

Reply:
> maybe the struct of NodeCache have this problem, its may cause a misunderstanding ？ I think its good for both of them coincide

from the current logic, nodecache does not have such a problem, but it will be better to change to the same processing logic